registry/registry.go

```diff
@@ -125,12 +127,17 @@ func newTransport(tlsConfig *tls.Config) *http.Transport {
 		KeepAlive: 30 * time.Second,
 	}
 
-	return &http.Transport{
-		Proxy:               http.ProxyFromEnvironment,
-		DialContext:         direct.DialContext,
-		TLSHandshakeTimeout: 10 * time.Second,
-		TLSClientConfig:     tlsConfig,
-		// TODO(dmcgowan): Call close idle connections when complete and use keep alive
-		DisableKeepAlives: true,
-	}
+	return otelhttp.NewTransport(
```

Comment:
I saw there's also a utility for this, not sure if that's better? (possibly it allows not having to import `otelhttp`, but leave that to to the utility); https://github.com/moby/moby/blob/ad6929339acd64ebae3fd55657d0dcf22895da86/vendor/github.com/containerd/containerd/tracing/tracing.go#L51-L59

Reply:
There is, but it takes an returns the whole `http.Client` instead of a `http.RoundTripper` so can't really use it